src/tox_travis/hooks.py

```diff
@@ -1,23 +1,28 @@
 """Tox hook implementations."""
 import tox
+from .utils import parser_cfg, config_cfg
 from .toxenv import default_toxenv, override_ignore_outcome
 from .after import travis_after_monkeypatch
 
 
-@tox.hookimpl
+@tox.hookimpl(trylast=True)
 def tox_addoption(parser):
     """Add arguments and override TOXENV."""
     parser.add_argument(
         '--travis-after', dest='travis_after', action='store_true',
         help='Exit successfully after all Travis jobs complete successfully.')
 
-    default_toxenv()
+    cfg = parser_cfg(parser)
+    if cfg:
+        default_toxenv(cfg)
 
 
 @tox.hookimpl
 def tox_configure(config):
     """Check for the presence of the added options."""
+    cfg = config_cfg(config)
+
     if config.option.travis_after:
-        travis_after_monkeypatch()
+        travis_after_monkeypatch(cfg)
 
-    override_ignore_outcome(config)
+    override_ignore_outcome(config, cfg)
```

Comment:
For this one, why not get the boolean `unignore_outcomes` here, and pass that to `override_ignore_outcome` as an arg.

I'd be tempted to do the same with `travis_after`, passing a `Section` or `env_requirements` list object down, so that the ini parsing is done here, which will make ini config errors happen earlier in the tox sequence.

Reply:
Way late, but FWIW I'm definitely considering this approach for a possible refactor. It mostly depends just how involved I need to get into the config. For unignore_outcomes, it's pretty simple. For toxenv, it's unfortunately a bit more complicated bacause of a backward compatibility concern. But it seems like a good approach in general.